pt_miniscreen/__main__.py

```diff
@@ -24,7 +25,62 @@
 @click_logging.simple_verbosity_option(logger)
 @click.version_option()
 def main() -> None:
-    app = App()
+    from pitop import Pitop
+
+    from .tile_group import TileGroup
+    from .tile_group_context import TileGroupContext, TileGroupContextManager
+    from .tiles import HUDMenuTile, SettingsMenuTile, SettingsTitleBarTile
+    from .tiles.settings_connection import SettingsConnectionMenuTile
+
+    logger.debug("Setting ENV VAR to use miniscreen as system...")
+    environ["PT_MINISCREEN_SYSTEM"] = "1"
+
+    miniscreen = Pitop().miniscreen
+
+    title_bar_height = 19
+
+    main = TileGroupContext(
+        tile_groups=[
+            TileGroup(
+                size=miniscreen.size,
+                menu_tile=HUDMenuTile(size=miniscreen.size),
+                title_bar_tile=None,
+            ),
+            TileGroup(
+                size=miniscreen.size,
+                title_bar_tile=SettingsTitleBarTile(
+                    size=(miniscreen.size[0], title_bar_height),
+                    pos=(0, 0),
+                ),
+                menu_tile=SettingsMenuTile(
+                    size=(
+                        miniscreen.size[0],
+                        miniscreen.size[1] - title_bar_height,
+                    ),
+                    pos=(0, title_bar_height),
+                ),
+            ),
+        ],
+    )
+    settings_connection = TileGroupContext(
+        name="settings_menu",
+        tile_groups=[
+            TileGroup(
+                size=miniscreen.size,
+                menu_tile=SettingsConnectionMenuTile(miniscreen.size),
+                title_bar_tile=SettingsTitleBarTile(
+                    size=(
+                        miniscreen.size[0],
+                        miniscreen.size[1] - title_bar_height,
+                    ),
+                    pos=(0, title_bar_height),
+                    text="Settings / Connection",
+                ),
+            ),
+        ],
+    )
+    context_manager = TileGroupContextManager(contexts=[main, settings_connection])
```

Comment:
Why do we need contexts managed like this? It seems to me a lot simpler to just have the settings menu tile have its own internal API for how to handle button press events, which the tile group should just route through to if active. Those button presses can be acted on to invoke "change menu" functionality, without needing to create any new tiles or groups, and also without creating new constructs like contexts.

Rather than creating a new group context for each child menu of the settings (defined in the `main` group) perhaps the settings tile group should be a `SettingsTileGroup` which understands all of the settings menu parent/child logic - we can emit events in the settings menu tile, and the group can

Reply:
I think this is a more "general" approach - instead of using a "tree-like" object definition where we need to re-use title bars and other stuff that's specific to the approach our app currently has (eg: top level menu with `hud` & `settings` `TileGroups`) we use contexts that don't care about the amount of `TileGroups`... 

One of the cool things I found about this approach is that each context can represent an app (eg: the "Settings Connection"), which converts into the visible context once the select button is pressed in the "Settings" Menu..

Anyway! I think we can come back to this at some point in the future